pkg/ast/spl/spl.peg

```diff
@@ -1565,7 +1565,11 @@ QuotedString <- '"' [^"]* '"' {
     return string(c.text), nil
 }
 
-UnquotedString <- ("*" / (!(MAJOR_BREAK / EOF) .))+ {
+UnquotedString <- ("true" / "false") {
+    boolValue, _ := strconv.ParseBool(string(c.text))
+    return boolValue, nil
+}
+/ ("*" / (!(MAJOR_BREAK / EOF) .))+ {
```

Comment:
The `UnquotedString` rules should return a string, so you'll need to move this logic to a different rule (or possibly make a new rule) that returns a boolean. Then you may need to update where that rule can appear as part of a different rule so that when we get a search like `isOpen=true` we parse `true` as a boolean but if we get something like `isOpen=tomorrow` we parse `tomorrow` as a string. In PEG grammars, the first option of a rule that matches succeeds, so this means if you had a rule like:
```
FieldAndValue <- field:FieldName "=" value:(Boolean / String) {
    ...
}
```
It's important that the order is `Boolean / String` because if it was `String / Boolean` then when you got `isOpen=true` it would try parsing `true` as a string and that would succeed, so it would never try to parse it as a boolean

Reply:
Got it, I will make the changes as soon as possible.

Reply:
@ParichayDidwania Any update on this PR?

Reply:
@nbhavana I will try and finish it by eod

Reply:
@nbhavana please check now